### PR TITLE
Coefficient weakness + room & trap price configurable

### PR DESCRIPTION
**Pull request by hwoarangmy**
To test weakness coef, I've changed the skirmish small test map to check it. Now, the big dragon is under the player starting position and it shouldn't flee until dead

### CODE REVIEW

File: source/rooms/Room.cpp [hunk 1]
```diff
@@ -365,25 +365,25 @@ int Room::costPerTile(RoomType t)
         return 0;
 
     case treasury:
-        return 25;
+        return ConfigManager::getSingleton().getRoomConfigInt32("TreasuryCostPerTile");
```
Comment (oyvindln) [inline source/rooms/Room.cpp]: I think we should really avoid using singletons like this.

**Comment by Bertram25** [comment]
Nothing to say. I'll test asap.

**Comment by Bertram25** [comment]
Works fine. We can merge IMHO.

**Comment by Bertram25** [comment]
I'll take the responsibility to merge now as I can't see anything really hurting here.

**Comment by hwoarangmy** [comment]
> I'll take the responsibility to merge now as I can't see anything really hurting here.

That's a huge responsability to merge my PRs ;-)

**Comment by Bertram25** [comment]
It's more to have someone to yell at, just in case. ;)

**Pull request merged**